packages/fgtclb/academic-projects/Classes/Upgrades/FlexFormUpgradeWizard.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FGTCLB\AcademicProjects\Upgrades;
+
+use TYPO3\CMS\Core\Database\Connection;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Install\Attribute\UpgradeWizard;
+use TYPO3\CMS\Install\Updates\DatabaseUpdatedPrerequisite;
+use TYPO3\CMS\Install\Updates\UpgradeWizardInterface;
+
+#[UpgradeWizard('academicProjects_flexFormUpgradeWizard')]
+final class FlexFormUpgradeWizard implements UpgradeWizardInterface
+{
+    private const MIGRATE_CONTENT_TYPES_LIST = [
+        'academicprojects_projectlist',
+        'academicprojects_projectlistsingle',
+    ];
+
+    private const MIGRATE_PLUGIN_SETTINGS = [
+        'setting.filter.options' => 'settings.hideFilter',
+        'settings.sorting.options' => 'settings.hideSorting',
+    ];
+
+    public function __construct(
+        private readonly ConnectionPool $connectionPool,
+    ) {}
+
+    public function getTitle(): string
+    {
+        return 'Migrate academic_projects FlexForm setting from "hideCompletedProjects" to "hideActiveState" selection.';
+    }
+
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function executeUpdate(): bool
+    {
+        $records = $this->getMigrationRecords();
+
+        foreach ($records as $record) {
+            // Nothing to upgrade
+            if ($record['pi_flexform'] === null || $record['pi_flexform'] === '') {
+                continue;
+            }
+
+            $flexFormData = GeneralUtility::xml2array($record['pi_flexform']);
+
+            if (isset($flexFormData ['data']['sDEF']['lDEF']['settings.hideCompletedProjects']['vDEF'])) {
+                // Migrate checkbox toggle to select option
+                if ($flexFormData['data']['sDEF']['lDEF']['settings.hideCompletedProjects']['vDEF'] === '1') {
+                    $flexFormData['data']['sDEF']['lDEF']['settings.activeState']['vDEF'] = 'active';
+                } else {
+                    $flexFormData['data']['sDEF']['lDEF']['settings.activeState']['vDEF'] = 'all';
+                }
+                unset($flexFormData ['data']['sDEF']['lDEF']['settings.hideCompletedProjects']);
+
+                // Rename settings
+                foreach (self::MIGRATE_PLUGIN_SETTINGS as $oldName => $newName) {
+                    if (isset($flexFormData['data']['sDEF']['lDEF'][$oldName]['vDEF'])) {
+                        $flexFormData['data']['sDEF']['lDEF'][$newName]['vDEF'] = $flexFormData['data']['sDEF']['lDEF'][$oldName]['vDEF'];
+                        unset($flexFormData['data']['sDEF']['lDEF'][$oldName]);
+                    }
+                }
```

Comment:
This loop over a list of settings is in the wrong level. It should be next to level startet in L53, and not within it.

In the current place, the other settings are only migrated if the hideCompletedProjects exits. Not a good thing.